src/providers/fcm.rs

```diff
@@ -27,18 +27,36 @@ impl FcmProvider {
 
 #[async_trait]
 impl PushProvider for FcmProvider {
+    #[instrument(name = "send_fcm_notification")]
     async fn send_notification(
         &mut self,
         token: String,
-        payload: MessagePayload,
+        body: PushMessageBody,
+        always_raw: bool,
     ) -> crate::error::Result<()> {
-        let s = span!(tracing::Level::DEBUG, "send_fcm_notification");
-        let _ = s.enter();
-
         let mut message_builder = MessageBuilder::new(self.api_key.as_str(), token.as_str());
 
-        let result = if payload.is_encrypted() {
-            message_builder.data(&payload)?;
+        // Sending `always_raw` encrypted message
+        let result = if always_raw {
+            info!("Sending raw encrypted message");
+            message_builder.data(&RawMessagePayload {
+                topic: body.topic.unwrap(),
```

Comment:
These unwraps scare me a little. I wonder if this could be refactored somehow so that the state of these values is tied to `always_raw`? E.g. `if let Some(...) = raw_values`?

Reply:
That makes sense, thx!
Now it's done with the [check](https://github.com/WalletConnect/echo-server/pull/281/files#diff-3f172be4d81562a96e145c96cc9cf4abcf9ec3db3c65d476d58f37b397133981R80-R86) and throw an appropriate error.

Reply:
I meant a way to avoid errors altogether. Like instead of passing `always_raw`, you check these in the calling function, and then pass the raw values through. So then you just check if there are raw values, rather than if `always_raw=true` and then asserting they are Somes a second time